runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/internal/DataflowGroupByKey.java

```diff
@@ -117,6 +140,20 @@ public PCollection<KV<K, Iterable<V>>> expand(PCollection<KV<K, V>> input) {
           "the keyCoder of a DataflowGroupByKey must be deterministic", e);
     }
 
+    PipelineOptions options = input.getPipeline().getOptions();
+    String gbekOveride = options.getGBEK();
+    if (!this.insideGBEK && gbekOveride != null && !gbekOveride.trim().isEmpty()) {
+      this.surroundsGBEK = true;
+      Secret hmacSecret = Secret.parseSecretOption(gbekOveride);
```

Comment:
just to confirm, is the access to actual secret required at pipeline expansion time? (read the code it seems the answer is no, which would be great) We have seen use cases that pipeline submission machine doesn't have access to certain resources (database for IO, etc) and causes some hurdle for users

Reply:
Yeah, that's right - only the runtime environment will access the secret